lib/Exporter/Writer/KnpSnappyWriter.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+/*
+ * This file is part of the Sonata package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Exporter\Writer;
+
+use Knp\Snappy\Pdf;
+
+/**
+ * Export data in Pdf format.
+ *
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+class KnpSnappyWriter implements WriterInterface
+{
+    /**
+     * @var string
+     */
+    protected $filename;
+    /**
+     * @var @resource
+     */
+    protected $tmpFile;
+
+    /**
+     * @var bool
+     */
+    protected $showHeaders;
+
+    /**
+     * @var bool
+     */
+    protected $showBorders;
+
+    /**
+     * @var array
+     */
+    protected $snappyOptions;
+
+    /**
+     * @var string
+     */
+    protected $wktmltopdfBinary;
+
+    /**
+     * @var int
+     */
+    protected $position;
+
+    /**
+     * @var Pdf
+     */
+    protected $snappy;
+
+    /**
+     * @throws \RuntimeException
+     *
+     * @param string $filename
+     * @param bool   $showHeaders
+     * @param bool   $showBorders
+     * @param array  $snappyOptions
+     * @param string $wkhtmltopdfBinary Path of the wkhtmltopdf binary
+     */
+    public function __construct($filename, $wkhtmltopdfBinary, $showHeaders = true, $showBorders = true, $snappyOptions = null)
+    {
+        $this->filename          = $filename;
+        $this->showHeaders       = $showHeaders;
+        $this->showBorders       = $showBorders;
+        $this->snappyOptions     = $snappyOptions;
+        $this->wkhtmltopdfBinary = $wkhtmltopdfBinary;
+        $this->position          = 0;
+
+        if (is_file($filename)) {
+            throw new \RuntimeException(sprintf('The file %s already exist', $filename));
+        }
+
+        if (!is_file($wkhtmltopdfBinary)) {
+            throw new \RuntimeException('Unable to find wkhtmltopdf binary');
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function open()
+    {
+        $this->tmpFile = fopen(sys_get_temp_dir().'/sonata_exporter_test.html', 'w', false);
```

Comment:
you need to use ``tmpfile`` function to get a unique file. Please read http://php.net/manual/fr/function.tmpfile.php